account_move_line_tax_editable/models/account_move_line.py

```diff
@@ -12,7 +12,56 @@ class AccountMoveLine(models.Model):
         string="Is tax data editable?", compute="_compute_is_tax_editable"
     )
 
+    tax_line_id = fields.Many2one(inverse="_inverse_tax_line_id")
+
     @api.depends("move_id.state")
     def _compute_is_tax_editable(self):
         for rec in self:
             rec.is_tax_editable = rec.move_id.state == "draft"
+
+    def _inverse_tax_line_id(self):
+        for rec in self:
+            repartition_type = rec.tax_repartition_line_id.repartition_type or "tax"
+            factor_percent = rec.tax_repartition_line_id.factor_percent or 100
+            has_account = bool(rec.tax_repartition_line_id.account_id)
+            if rec.move_id.move_type in ("out_refund", "in_refund"):
+                rec.tax_repartition_line_id = rec.find_suitable_tax_repartition_line_id(
+                    rec.tax_line_id.refund_repartition_line_ids,
+                    repartition_type,
+                    factor_percent,
+                    has_account,
+                )
+            else:
+                rec.tax_repartition_line_id = rec.find_suitable_tax_repartition_line_id(
+                    rec.tax_line_id.invoice_repartition_line_ids,
+                    repartition_type,
+                    factor_percent,
+                    has_account,
+                )
+
+    def find_suitable_tax_repartition_line_id(
+        self,
+        repartition_lines,
+        repartition_type="tax",
+        factor_percent=100,
+        has_account=True,
+    ):
+        self.ensure_one()
+        lines = repartition_lines.filtered(
```

Comment:
Why not simply put this code out of the `if` / `else` of the previous method? Refactor code into a method worths when there are multiple calls from several methods, but if the call comes from the same method, you can integrate this code in the method itself. But if you extract it, put this method as private.

Reply:
Oh, right, I didn't get it before.

Reply:
Done